### ISSUE TITLE
Robust Incremental Structure-from-Motion with Hybrid Features

### ISSUE THREAD

**Issue by reynoldscem** [cvg/limap#85]
Hi,

Really great work. Am I to understand the plan is to release the code from the ECCV paper as an update to this repo? I’m very interested in trying it out.

If so, when do you expect the code release to roughly be?

Thanks

**Comment by B1ueber2y** [comment]
Hey. Thanks for the question. Yes I can confirm that this is the plan. The rough timeline is to release by the end of this month. And we will make sure that the code is released before the conference (end of September).

**Comment by reynoldscem** [comment]
Fantastic! Enjoy the conference and congrats

**Comment by Idea-in-Dream** [comment]
﻿I am looking forward to your work very much

**Comment by B1ueber2y** [comment]
Due to overloaded schedule for my internship and the actively moving target of COLMAP (we plan to support the new code release with the newest COLMAP version), we will have to delay the code release to ensure that it is as good and flexible as possible. Sorry for that and stay tuned for our updates!

**Comment by RaghavM11** [comment]
Hey thanks for update!

By any chance, have u done ablation studies on synthetic dataset like tartan air using recent approach? 
(would have been cool ig)

I used limap 1.0 on a lot of synthetic dataset..but as expected it broke where i expected it to..

Just curious about your opinion.

Thanks again!

**Comment by reynoldscem** [comment]
@B1ueber2y No worries! Thanks for putting out such great work

**Comment by Vilour** [comment]
> Hey. Due to overloaded schedule for my internship and the actively moving target of COLMAP (we plan to support the new code release with the newest COLMAP version), we will have to delay the code release to ensure that it is as good and flexible as possible. Sorry for that and stay tuned for our updates!

Now is 2025 and we are still anticipating 😃

**Comment by Idea-in-Dream** [comment]
@B1ueber2y is this still in the pipeline for release? I am looking forward to your work very much

**Comment by B1ueber2y** [comment]
> [@B1ueber2y](https://github.com/B1ueber2y) is this still in the pipeline for release? I am looking forward to your work very much

Thanks for asking. Actually I think it s the right time to make some updates in this thread. The short answer is yes, but unfortunately it would take longer.

Internally, the whole limap has been completely rewritten to align with the development flow of COLMAP, and this whole new limap will be released by around March next year. This is possible only due to tons of improvements on the colmap side the past year to make the interface more extensible, both in C++ and in Python. 

There are several benefits for this whole migration:

1) shared utilities directly from the COLMAP progression, such as faster solvers, more robust two-view geometry, and even more importantly, the new multi-sensor support. 
2) In-place optimization rather than copying back and forth between the limap ``ImageCollection`` and the colmap ``Reconstruction``. 
3) cleaner data flow with the abstraction of colmap database and structure database. 

Thus, this migration is very necessary to keep the future of limap project maintainable and adaptable with COLMAP (and also GLOMAP). 

This being said, the whole limap sfm described in this ECCV paper is also rewritten, and we are working on a much faster bundle adjustment alternative, which will eventually make running sfm with hybrid features even more practical than described in the paper. We plan to write an academic paper on some of the new tech contributions so this will take some time until being finished. The new limap will also have several exciting new features such as faster wireframe and planar grouping, so please stay tuned and sorry for keeping you wait more time.

**Comment by Yancccccc** [comment]
@B1ueber2y is this still in the pipeline for release? I am looking forward to your work very much

**Comment by B1ueber2y** [comment]
> [@B1ueber2y](https://github.com/B1ueber2y) is this still in the pipeline for release? I am looking forward to your work very much

Yes. The target timeline is end of March. We may also include a global sfm alternative with lines and structures.

**Comment by B1ueber2y** [comment]
The SfM system is released in https://github.com/cvg/limap/pull/155 along with the new paper (coming out on arxiv tomorrow). The system consistently improves point-alone baselines particularly in structured indoor environments. 

A small note here: the release uses the upstreamed hybrid ransac (pnpl and pnplf) module in poselib, so currently does not include the uncertainty-aware localization.

**Comment by reynoldscem-oculo** [comment]
Amazing, thank you! Congrats on the paper
